io/net/irqbalance.py

```diff
@@ -0,0 +1,437 @@
+        return self.irq_number
+
+    def get_ping_process_pid(self):
+        """
+        Funtion to get the process ID of ping flood.
+
+        :returns : Process PID number that initated by ping flood command.
+        :rtype : int
+        """
+        cmd = (
+            f"ps -ef | grep '[p]ing -I "
+            f"{self.interface} {self.peer_ip} -c {self.ping_count} -f' "
+            f"| awk '{{print $2}}' | head -1"
+        )
+        process_pid = process.system_output(cmd, shell=True,
+                                            ignore_status=True,
+                                            sudo=True).decode("utf-8")
+        if not process_pid:
+            self.log.debug(f"No more process PID avaialable")
+            return False
+        return process_pid
+
+    def compare_range_strings(self, range_str1, range_str2):
+        '''
+        Function to compare and match cpu range of string format.
+        :rtype : boolean value.
+        '''
+        range_pattern = r'(\d+)-(\d+)'
+        match1 = re.match(range_pattern, range_str1)
+        match2 = re.match(range_pattern, range_str2)
+
+        if match1 and match2:
+            start1, end1 = int(match1.group(1)), int(match1.group(2))
+            start2, end2 = int(match2.group(1)), int(match2.group(2))
+
+            return start1 == start2 and end1 == end2
+        return False
+
+    def cpu_range_validation(self):
+        '''
+        Funtion to validate the assinged CPU's by script.
+        '''
+        self.irq_affinity = '-'.join([str(self.cpu_range[0]),
+                                      str(self.cpu_range[-1])])
+        cmd = f'/proc/irq/{self.irq_number}/smp_affinity_list'
+        self.system_affinity = genio.read_file(cmd)
+        time.sleep(25)
+        if not self.compare_range_strings(self.irq_affinity,
+                                          self.system_affinity):
+            self.fail(f'The smp_affinity_list does not matches to affinity'
+                      ' that was set by script, Please check the logs')
+
+    def grep_module_interrupts(self):
+        '''
+        Funtion to filter all interrupts along associated CPU's of device.
+        '''
+        cmd = f'head -n 1 /proc/interrupts &&' \
+              f' grep {self.interface_type} /proc/interrupts'
+
+        process.run(cmd, shell=True, ignore_status=True
+                    ).stdout.decode().strip()
+
+    def taskset_cpu_validation(self):
+        '''
+        Function to validate the CPU number set by "taskset" command.
+        Returns : <int> value set by script.
+        '''
+        cmd = "ps -o psr -p %s | awk 'NR>1 {print $1}'" % (
+            self.get_ping_process_pid()
+        )
+        cpu_by_script = process.system_output(cmd, shell=True,
+                                              ignore_status=True,
+                                              sudo=True).decode("utf-8")
+        return int(cpu_by_script)
+
+    def dd_process_pid(self):
```

Comment:
I think you can change this to disk_taskset_cpu_validation()

Reply:
better name would be get_dd_process_pid   as this is not validating any cpu task set

Reply:
@FarooqAbdulla01 rename please

Reply:
@abdhaleegit @Naresh-ibm 
renamed to get_dd_process_pid

along with added support for taskset_cpu_validation for block based interfaces.